File: modules/sql/app/model/querycorrectors/change/DeleteCorrector.java
package model.querycorrectors.change;

import java.util.Arrays;
import java.util.List;
import java.util.Map;

import javax.inject.Singleton;

import model.matching.MatchingResult;
import model.querycorrectors.ColumnMatch;
import net.sf.jsqlparser.expression.Expression;
import net.sf.jsqlparser.schema.Column;
import net.sf.jsqlparser.schema.Table;
import net.sf.jsqlparser.statement.delete.Delete;

@Singleton
public class DeleteCorrector extends ChangeCorrector<Delete> {

  public DeleteCorrector() {
    super("DELETE");
  }

  @Override
  protected MatchingResult<Column, ColumnMatch<Column>> compareColumns(Delete userQuery,
      Map<String, String> userTableAliases, Delete sampleQuery, Map<String, String> sampleTableAliases) {
    return null;
  }

  @Override
  protected List<String> getTableNames(Delete userQuery) {
    return Arrays.asList(userQuery.getTable().getName());
  }

  @Override
  protected List<Table> getTables(Delete query) {
    return query.getTables();
  }

  @Override
  protected Expression getWhere(Delete query) {
    return query.getWhere();
  }
  
}

File: app/model/tools/sql/DeleteCorrector.scala
package model.tools.sql

import net.sf.jsqlparser.expression.Expression
import net.sf.jsqlparser.schema.Table
import net.sf.jsqlparser.statement.Statement
import net.sf.jsqlparser.statement.delete.Delete

import scala.jdk.CollectionConverters._
import scala.util.{Failure, Success, Try}

object DeleteCorrector extends ChangeCorrector("DELETE") {

  override type Q = Delete

  override protected val sqlExecutionDAO: SqlExecutionDAO[Delete] = DeleteDAO

  override protected def getTables(query: Q): Seq[Table] = query.getTables.asScala.toSeq

  override protected def getWhere(query: Q): Option[Expression] = Option(query.getWhere)

  override protected def checkStatement(statement: Statement): Try[Delete] = statement match {
    case q: Delete => Success(q)
    case other     => Failure(WrongStatementTypeException(queryType, gotten = other.getClass.getSimpleName))
  }

  override protected def performAdditionalComparisons(userQuery: Delete, sampleQuery: Delete): AdditionalComparison = AdditionalComparison(None, None)

}
